### ISSUE TITLE
[torch.compile] autograd.Function graph breaks on operations on ctx attrs

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#106390]
### 🐛 Describe the bug

```py
import torch
import torch._dynamo

class Foo(torch.autograd.Function):
    @staticmethod
    def forward(ctx, x):
        ctx.save_for_backward(x)
        ctx.x_dim = x.dim
        return x.sin()
    @staticmethod
    def backward(ctx, grad):
        x, = ctx.saved_tensors
        return grad * x.cos() * ctx.x_dim

@torch.compile(backend='aot_eager', fullgraph=True)
def f(x):
    return Foo.apply(x)

x = torch.randn([1], requires_grad=True)
f(x)
```
fails with:
```
Unsupported: call_function BuiltinVariable(mul) [TensorVariable(), GetAttrVariable(TensorVariable(), dim)] {}

from user code:
   File "/raid/rzou/pt/debug-cpu/torch/_dynamo/variables/misc.py", line 241, in trampoline_autograd_bwd
    return fn_cls.backward(*args, **kwargs)
  File "<ipython-input-10-2818c59f5f2c>", line 13, in backward
    return grad * x.cos() * ctx.x_dim

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information
```

### Versions

main

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

**Comment by voznesenskym** [comment]
Debugged w/ @anijain2305 and @yanboliang - the repro is incorrect-  `dim` is a method. Changing the repro to `ndim` or `dim()` works and does not raise.

**Comment by anijain2305** [comment]
@zou3519 back to you.

**Comment by ezyang** [comment]
seems like we should... try a bit harder to give a good error message here lol

What is particularly confusing is that the eager program doesn't fail, because backward() is never called.
